src/node/snapshotter.h

```diff
@@ -248,16 +258,34 @@ namespace ccf
       std::lock_guard<std::mutex> guard(lock);
 
       CCF_ASSERT_FMT(
-        idx >= next_snapshot_indices.back(),
+        idx >= next_snapshot_indices.back().idx,
         "Committable seqno {} < next snapshot seqno {}",
         idx,
-        next_snapshot_indices.back());
+        next_snapshot_indices.back().idx);
+
+      bool forced = store->flag_enabled(
+        kv::AbstractStore::Flag::SNAPSHOT_AT_NEXT_SIGNATURE);
+
+      consensus::Index last_unforced_idx = last_snapshot_idx;
+      for (size_t i = next_snapshot_indices.size() - 1;
+           i < next_snapshot_indices.size();
```

Comment:
This condition seems weird if we iterate through the list backwards, no?

Reply:
Yeah, but `!= SIZE_MAX` is equally weird...